src/js/carousel.js

```diff
@@ -0,0 +1,114 @@
+(function (window) {
+  function createDots() {
+    var existingDots = this.$carousel.getElementsByClassName('dots');
+    if (existingDots.length) {
+      this.$carousel.removeChild(existingDots[0]);
+    }
+    var dots = document.createElement('ul');
+    dots.className = 'dots';
+    var items = this.$carousel.getElementsByTagName('ul')[0];
+    this.$count = items.childElementCount;
+    for (var i = 0; i < items.childElementCount; i += 1) {
+      var dot = document.createElement('li');
+      dot.dataset.index = i;
+      dots.appendChild(dot);
+    }
+    this.$carousel.appendChild(dots);
+  }
+
+  function selectView(index) {
+    var scrollPosition = index * this.$width;
+    var items = this.$carousel.firstElementChild;
+    items.scrollTo(scrollPosition, 0);
+  }
+
+  function unselectAllDots() {
+    var dots = this.$carousel.lastElementChild;
+    for (var i = 0; i < dots.childElementCount; i += 1) {
+      dots.children[i].className = '';
+    }
+  }
+
+  function highlightDot(index) {
+    this.unselectAllDots();
+    var dots = this.$carousel.lastElementChild;
+    for (var i = 0; i < dots.childElementCount; i += 1) {
+      if (i === index) {
+        dot = dots.children[i].className = 'active';
+      }
+    }
+  }
+
+  function update() {
+    this.selectView(this.$selectedView);
+    this.highlightDot(this.$selectedView);
+  }
+
+  function _onDotClick(e) {
```

Comment:
Another self convention to remember me that this function is not directly bound to a _class_ property. And since it is a function that is declared in the closure, it is like a private function. So I like to give the name a hint.